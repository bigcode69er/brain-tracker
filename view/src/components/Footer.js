import React, { useState } from 'react';
import { Footer, Tile, Title, Content, Button } from 'rbx';

// Named Credits
export default function() {
  const [region, setRegion] = useState(0);

  // TODO: Add country information
  const information = [
    {
      country: 'United States',
      name: 'National Suicide Prevention Lifeline',
      number: '1-800-273-8255',
      desc:
        'The Lifeline provides 24/7, free and confidential support for people in distress, prevention and crisis resources for you or your loved ones, and best practices for professionals.'
    },
    { country: 'Canada', number: 'Soon', desc: 'soon' },
    { country: 'Mexico', number: 'Soon', desc: 'soon' },
    { country: 'United Kingdom', number: 'Soon', desc: 'soon' }
  ];

  return (
    <Footer>
      <Tile kind="ancestor">
        <Tile kind="parent">
          <Title size={6}>Credits</Title>
        </Tile>
        <Tile kind="parent">
          <Title size={6}>Made by Owais</Title>
        </Tile>
        {/* Crisis Line */}
        <Tile kind="parent" vertical>
          <Tile kind="child">
            <Title size={6}>{information[region].name}</Title>
            <Title size={6} subtitle>
              {information[region].number}
            </Title>
            <Content size="small">{information[region].desc}</Content>
          </Tile>
          {/* Country Toggles */}
          <Tile kind="child">
            <Button.Group>
              {information.map((country, index) => {
                return (
                  <Button
                    rounded
                    outlined
                    color="dark"
                    size="small"
                    onClick={() => setRegion(index)}
                    key={index}
                  >
                    {country.country}
                  </Button>
                );
              })}
            </Button.Group>
          </Tile>
        </Tile>
      </Tile>
    </Footer>
  );
}
