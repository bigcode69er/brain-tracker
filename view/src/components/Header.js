import React from 'react';
import { Hero, Container, Title } from 'rbx';
import { Link } from 'react-router-dom';

export default function Header() {
  return (
    <Hero color="success">
      <Hero.Body>
        <Container>
          <Title>
            <Link className="title-link" to="/">
              Brain Tracker
            </Link>
          </Title>
        </Container>
      </Hero.Body>
    </Hero>
  );
}
