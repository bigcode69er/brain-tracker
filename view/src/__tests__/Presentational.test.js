import React from 'react';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import Footer from '../components/Footer';

describe('Header', () => {
  test('Snapshot Renders', () => {
    const component = TestRenderer.create(
      <Router>
        <Header />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Footer', () => {
  test('Snapshot Renders', () => {
    const component = TestRenderer.create(<Footer />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
