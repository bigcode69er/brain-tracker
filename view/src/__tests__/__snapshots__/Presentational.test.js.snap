// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Renders 1`] = `
<div
  className="footer"
>
  <div
    className="container"
  >
    <div
      className="content"
    >
      Made by Owais Jamil
    </div>
  </div>
</div>
`;

exports[`Header Snapshot Renders 1`] = `
<section
  className="hero is-success"
>
  <div
    className="hero-body"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        <a
          className="title-link"
          href="/"
          onClick={[Function]}
        >
          Brain Tracker
        </a>
      </h1>
    </div>
  </div>
</section>
`;
